### ISSUE TITLE
Publish sidebar focus loss and modal behavior

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#45417]
### Description

When using the keyboard and closing the publish sidebar, focus is lost. 

A focus loss is a very confusing and frustrating experience for keyboard users, as they are forced to  start navigating the user interface from the beginning of the document.

Also, tabbing with the Tab key is constrained within the publish sidebar, which is good. However, screen reader users can navigate outside the sidebar by arrowing or by using other screen readers navigation mechanisms. If we want all users to stay within the publish sidebar while they're in the publishing flow, the sidebar should have a modal behavior. This can be accomplished by adding an `aria-modal=true` attribute.

### Step-by-step reproduction instructions

Test 1:
- Edit a post draft or create a new one.
- Navigate to the Publish button by using the keyboard Tab key.
- Press Enter to open the Publish sidebar.
- Press Tab to move to the Cancel button and press Enter.
- The Publish sidebar closes.
- Press the Tab key again and observe focus is now on the browser interface.
- See the first animated GIF that illustrates the focus loss. 

Test 2:
- Edit a post draft or create a new one.
- Navigate to the Publish button by using the keyboard Tab key.
- Press Enter to open the Publish sidebar.
- Press Enter again to Publish the post. 
- The sidebar switches to the 'post publish' flow. Focus is now on the link to view the post.
- Press Shift + Tab to move to the 'Close panel' X button. Press Enter to close the sidebar.
- Press the Tab key again and observe focus is now on the browser interface.
- See the first animated GIF that illustrates the focus loss. 

Test 3, Optional:
- Use a screen reader e.g. VoiceOver with Safari on macOS.
- Navigate using the keyboard and press the Publish button to open the Publish sidebar.
- Press Tab multiple times to navigate within the sidebar and observe tabbing is constrained within the sidebar, so far so good.
- Use the screen reader arrow keys navigation *with VoiceOver it's Control+Option+Right arrow or Control+Option+Left arrow).
- Observe with arrows navigation you can navigate away from the Publish sidebar.

### Screenshots, screen recording, code snippet

Test 1 animated GIF:

![focus-loss](https://user-images.githubusercontent.com/1682452/199014194-4bbfc93d-0009-424e-b900-f5f09969060c.gif)

Test 2 animated GIF:

![focus-loss2](https://user-images.githubusercontent.com/1682452/199014242-1458e1ff-ba01-47db-bbe0-c06302213733.gif)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
Just submitted a PR for the focus loss issue. Will split the modal behavior issue to a separate PR.
